### ISSUE TITLE
CatBoostClassifier `classes_` attribute is empty after fitting

### ISSUE THREAD

**Issue by zoj613** [catboost/catboost#1017]
Problem: CatBoostClassifier `classes_` attribute is empty after fitting. This causes problems when using the classifier with methods that require a non-empty `classes_` attribute.
steps to reproduce:
```python
from catboost import CatBoostClassifier
from sklearn.datasets import make_classification

X, y = make_classification(100, 10)

cat = CatBoostClassifier(verbose=0)
cat.fit(X, y)
print(cat.classes_)  # empty list
```

catboost version: 0.17.1
Operating System: Ubuntu 18.04.3 LTS
CPU: Intel® Core™ i7-8750H CPU @ 2.20GHz × 12 
GPU: GeForce GTX 1060 with Max-Q Design/PCIe/SSE2

**Comment by zoj613** [comment]
The lack of issue support on this project is appalling.

**Comment by annaveronika** [comment]
We are currently working on the fix for this problem, the code is in review, please be patient.

**Comment by zoj613** [comment]
> We are currently working on the fix for this problem, the code is in review, please be patient.

thank you for notifying me.

**Comment by zoj613** [comment]
Whats the status on the fix for this issue @annaveronika ?

**Comment by annaveronika** [comment]
Still in progress, most probably will come out in one of the next minor versions.

**Comment by zoj613** [comment]
ping @annaveronika

**Comment by andrey-khropov** [comment]
This first part of the issue was fixed by commit https://github.com/catboost/catboost/commit/0b4a4353887455b1042aa04ccb98226ee8c02097 and released in 0.18 (was omitted from release notes by oversight). After that commit ``classes_`` attribute has been non-empty but always contains strings which is a problem for some sklearn interfaces (like #999 for example).

So the second part is proper handling of different label types that was introduced in commit https://github.com/catboost/catboost/commit/f2d4cc2aa3b9a6421b731dd97634ec2c3191c181.

This commit will be in the new release within a day.
